File: Đức Trần/12_đệ_quy_quay_lui/dequy.cpp
#include <bits/stdc++.h>

using namespace std;

void hello(int n) {
    if (n <= 0)
        return;
    cout << "Hello " << n << endl;
    hello(n - 1);
    hello(n - 2);
}

int main() {
    hello(3);

    return 0;
}

File: Cô Giáo Thanh Vân/15_đệ_quy_quay_lui/dequy.py
def hello(n):
    if n == 0:
        return
    print("Hello", n)
    hello(n-1)

def gt(n):
    if n == 0:
        return 1
    return n * gt(n - 1)

# 1 * 2 * 3 * ... * n
# gt(n): Tính giai thừa của n
# n! = n * (n - 1)!
# gt(n) = n * gt(n-1)